yarn-project/circuit-types/src/l2_block.ts

```diff
@@ -264,19 +260,7 @@ export class L2Block {
    */
   toBuffer() {
     return serializeToBuffer(
-      this.header.globalVariables,
-      // TODO(#3868)
-      AppendOnlyTreeSnapshot.empty(), // this.startNoteHashTreeSnapshot,
-      AppendOnlyTreeSnapshot.empty(), // this.startNullifierTreeSnapshot,
-      AppendOnlyTreeSnapshot.empty(), // this.startContractTreeSnapshot,
-      AppendOnlyTreeSnapshot.empty(), // this.startPublicDataTreeSnapshot,
-      AppendOnlyTreeSnapshot.empty(), // this.startL1ToL2MessageTreeSnapshot,
-      this.header.lastArchive,
-      this.header.state.partial.noteHashTree,
-      this.header.state.partial.nullifierTree,
-      this.header.state.partial.contractTree,
-      this.header.state.partial.publicDataTree,
-      this.header.state.l1ToL2MessageTree,
+      this.header,
       this.archive,
       this.newCommitments.length,
       this.newCommitments,
```

Comment:
Would be nice for a future PR to have the body done as a separate struct as well 👍